### PR TITLE
WB-1170: Add Storybook interaction testing to snapshot tooltip stories correctly

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

NOTE: This PR is based on #1599.

Introduces Storybook interaction testing to be able to create stable
tooltip snapshots in Chromatic.

This feature will help us to introduce more interaction tests that will
be used in Chromatic where we need to use focus/hover behaviors to
verify that some of the components are working correctly (e.g. Modals,
Popovers, Dropdowns, etc).

Issue: WB-1170

## Test Plan:

Verify that the tooltip stories are correctly snapshotted in Chromatic.

### CODE REVIEW

File: packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js [hunk 1]
```diff
@@ -50,16 +54,34 @@ Default.args = {
...
+    // Triggers the hover state
+    await waitFor(async () => {
+        await userEvent.hover(canvas.getByText("some text"));
+    });
+
+    // Assert
+    await expect(
+        await canvas.findByText("This is a text tooltip on the top"),
+    ).toBeInTheDocument();
+};
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js:72]: Nice!

File: packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js [hunk 2]
```diff
@@ -50,16 +54,34 @@ Default.args = {
...
+Default.play = async ({canvasElement}) => {
+    // Arrange
+    // NOTE: Using `body` here to work with React Portals.
+    const canvas = within(canvasElement.ownerDocument.body);
+
+    // Act
+    // Triggers the hover state
+    await waitFor(async () => {
+        await userEvent.hover(canvas.getByText("some text"));
+    });
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js:66]: **question:** Does it take a while for `"some text"` to show up?  Would the following work?
```
const text = await canvas.findByText("some text");
await userEvent.hover(text);
```

File: packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js [hunk 3]
```diff
@@ -102,6 +141,10 @@ export const AnchorInScrollableParent: StoryComponentType = () => (
 );
 
 AnchorInScrollableParent.parameters = {
+    // Disable Chromatic because it only shows the trigger element.
+    chromatic: {
+        disableSnapshot: true,
+    },
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js:147]: **question:** How does Chromatic know when to take the snapshot of the stories using `.play()`?  Does it wait until `.play()` finishes before snapshotting?

**Review by kevinbarabash — APPROVED** [review]
LGTM.  Thanks for snapshotting the tooltip component.  Looking forward to see snapshots for more of our dynamic components in the future.

File: packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js (hunk 3 above)
Comment (jandrade) [reply packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js:147]: That's correct! https://www.chromatic.com/docs/hoverfocus#make-your-stories-interactive

> Chromatic awaits play function execution before taking a snapshot.

File: packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js (hunk 2 above)
Comment (jandrade) [reply packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js:66]: hmmm good point! that should work. Let me change it.

Comment (jandrade) [reply packages/wonder-blocks-tooltip/src/components/__docs__/tooltip.stories.js]: It worked, thanks for the suggestion 🎉

**Pull request merged**